src/AbstractFixer.php

```diff
@@ -34,7 +43,7 @@ public function __construct()
     {
         if ($this instanceof ConfigurableFixerInterface) {
             try {
-                $this->configure(null);
+                $this->configure(array());
```

Comment:
memory issue could be figured out. let us focus on first one first.

> It made header_comment configuration throw errors because it depends on the WhitespacesFixerConfig instance of the fixer, which can change between configure() calls.

It should not be done that way.
It's possible to run both methods - `configure` and `setWhitespacesConfig`.
If one would call: `setWhitespacesConfig`, `configure` - it works.
If one would call: `setWhitespacesConfig`, `configure`, `setWhitespacesConfig`, `configure` - it works.
If one would call `configure`, `setWhitespacesConfig` - it doesn't work, because configuration is set to outdated, previous whitespaces - one need to call `configure` all after all times he would call `setWhitespacesConfig` for `header_comment` Fixer.
I hope I show you the shortcoming of current implementation.

Both methods must be independent. I think you must remove that normalizer and call it not during applying configuration , but during using the value from config